### ISSUE TITLE
Environment setup

### ISSUE THREAD

**Issue by JimmyLaurent** [zoogie/old-browserhax#2]
Hi Zoogie,

First, good job ! Your browserhaxs are a good starting point for browser exploitation on 3DS on recent firmwares.

Would you mind sharing your environment setup ? like are you debugging in citra or luma3DS ? which loader are you using to load 3DS binaries in ghidra/ida ? etc..

Thanks !

**Comment by zoogie** [comment]
I use luma3ds and IDA. I dump the .code and .cro's from godmode9 and load it to 0x100000 in IDA. 

I also used luma's debugger + IDA 7.0 remote gdb stub to find out where the oss.cro and webkit.cro are located in virtual memory, and appended those to the above codebin at their proper relative addresses. I don't use dynamic debugging though, it's just to find out where things are, and then use feedback from the luma crashes to get control of registers. Check out MrNbaYohs blogs I linked in the Thanks section on exploiting webkit use after frees for more info on actual exploit techniques.
